### DISCUSSION TITLE
Add support for genesisID

### DISCUSSION THREAD

**Discussion by lrettig** [athenavm/athena#284] [Ideas]
Athena currently doesn't support this at all. We'd need the wallet template's verify method to support it, which begs the question, how does it know the genesisID in the first place? A few possibilities:

- the host passes it into the verify() method on each call
- it gets added to the wallet immutable state (i.e., it becomes part of the Wallet struct)
- it gets passed into the VM at instantiation time, and the VM knows to add it to every hash operation (a bit like what we do in go-sm)

**Comment by poszu** [comment]
I think it doesn't require any changes on the VM side to support signing TX data prefixed with the genesis ID. The signed TX data is passed as opaque bytes to the `verify()` as method inputs. The host side could prepend the genesis ID
